engine/src/test/java/io/zeebe/engine/processor/StreamProcessorInconsistentPositionTest.java

```diff
@@ -0,0 +1,123 @@
+import io.zeebe.util.sched.testing.ActorSchedulerRule;
+import java.util.concurrent.CountDownLatch;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+
+public final class StreamProcessorInconsistentPositionTest {
+
+  private final TemporaryFolder tempFolder = new TemporaryFolder();
+  private final AutoCloseableRule closeables = new AutoCloseableRule();
+  private final ControlledActorClock clock = new ControlledActorClock();
+  private final ActorSchedulerRule actorSchedulerRule = new ActorSchedulerRule(clock);
+
+  @Rule
+  public RuleChain ruleChain =
+      RuleChain.outerRule(tempFolder).around(actorSchedulerRule).around(closeables);
+
+  private StreamProcessingComposite firstStreamProcessorComposite;
+  private StreamProcessingComposite secondStreamProcessorComposite;
+  private TestStreams testStreams;
+
+  @Before
+  public void setup() throws Exception {
+
+    testStreams = new TestStreams(tempFolder, closeables, actorSchedulerRule.get());
+
+    final var dataDirectory = tempFolder.newFolder();
+
+    final AtomixLogStorageRule logStorageRule = new AtomixLogStorageRule(tempFolder, 1);
+    logStorageRule.open(
+        b ->
+            b.withDirectory(dataDirectory)
+                .withMaxEntrySize(4 * 1024 * 1024)
+                .withMaxSegmentSize(128 * 1024 * 1024));
+
+    testStreams.createLogStream(getLogName(1), 1, logStorageRule);
+    testStreams.createLogStream(getLogName(2), 2, logStorageRule);
+
+    firstStreamProcessorComposite =
+        new StreamProcessingComposite(testStreams, 1, DEFAULT_DB_FACTORY);
+    secondStreamProcessorComposite =
+        new StreamProcessingComposite(testStreams, 2, DEFAULT_DB_FACTORY);
+  }
+
+  @Test
+  public void shouldNotStartOnInconsistentLog() {
+    // given
+    final var position =
+        firstStreamProcessorComposite.writeWorkflowInstanceEvent(ELEMENT_ACTIVATING);
+    final var secondPosition =
+        firstStreamProcessorComposite.writeWorkflowInstanceEvent(ELEMENT_ACTIVATED);
+    waitUntil(
+        () ->
+            new RecordStream(testStreams.events(getLogName(2)))
+                .onlyWorkflowInstanceRecords()
+                .withIntent(ELEMENT_ACTIVATED)
+                .exists());
+
+    final var otherPosition =
+        secondStreamProcessorComposite.writeWorkflowInstanceEvent(ELEMENT_COMPLETING);
+    final var otherSecondPosition =
+        secondStreamProcessorComposite.writeWorkflowInstanceEvent(ELEMENT_COMPLETED);
+    waitUntil(
+        () ->
+            new RecordStream(testStreams.events(getLogName(2)))
+                .onlyWorkflowInstanceRecords()
+                .withIntent(ELEMENT_COMPLETED)
+                .exists());
+
+    assertThat(position).isEqualTo(otherPosition);
+    assertThat(secondPosition).isEqualTo(otherSecondPosition);
+
+    // when
+    final CountDownLatch recoveredLatch = new CountDownLatch(1);
```

Comment:
Since on recovery we switch the processor phase to `Phase.PROCESSING`, would comparing phases only not be enough? What additional guarantees does the latch provide?

Reply:
It is more a remaining because I checked before whether I call this method. I kept that to verify that after we failed we haven't called it, but probably not necessary.